### PR TITLE
Add FreeRTOS

### PR DESCRIPTION
**Pull request by dhylands**
This implements the producer/consumer from the freertos demo and also implements the LED blinker as a task too.

### CODE REVIEW

File: cc3200-sys/src/lib.rs [hunk 1]
```diff
@@ -14,6 +14,7 @@ extern {
   // From board.c
   pub fn board_init();
   pub fn console_putchar(char: i8);
+  pub fn test();
```
Comment (fabricedesre) [inline cc3200-sys/src/lib.rs]: Leftover? I don't see the corresponding change in board.c

File: src/main.rs [hunk 2]
```diff
@@ -11,10 +11,20 @@
...
+
+extern crate alloc;
+extern crate freertos_rs;
+extern crate freertos_alloc;
 
 mod cc3200;
 use cc3200::{Board, Utils, LedEnum, LedName};
 
+use alloc::arc::Arc;
+use freertos_rs::{ CurrentTask, Duration, Task, Queue };
```
Comment (fabricedesre) [inline src/main.rs]: Nit: ::{CurrentTask, ..., Queue}
I usually use rustfmt to force formatting. Sometimes it's a bit over the top like if you use it on isr_vectors.rs

**Review by fabricedesre — APPROVED** [review]
minor nits, but that looks great!

File: src/main.rs (hunk 2 above)
Comment (dhylands) [reply src/main.rs]: ok - on foxlink we wanted the spaces, so I was just following that convention.

Comment (fabricedesre) [reply src/main.rs]: Yep, I think we did a mistake there in foxlink.

**Comment by dhylands** [comment]
Cleaned up and ran cargo fmt (except on isr_vector.rs)

**Pull request merged**